### ISSUE TITLE
Possible mistake in Deep Q Learning Space Invaders notebook

### ISSUE THREAD

**Issue by karolisjan** [simoninithomas/Deep_reinforcement_learning_Course#51]
Hey. Shouldn't  `self.Q = tf.reduce_sum(tf.multiply(self.output, self.actions_))` in DQN class be `self.Q = tf.reduce_sum(tf.multiply(self.output, self.actions_), axis=1)`, i.e. reduced along columns so that the output length of `self.Q` is equal to the batch size? If not then `self.Q` will be a scalar while `self.target_Q` will be a vector of  batch size length.

**Comment by ali-ehsan** [comment]
@karolisjan I agree.
